python/cudf/cudf/core/index.py

```diff
@@ -854,7 +854,6 @@ def __init__(self, values, **kwargs):
                 pd.Categorical(values, categories=values)
             )
         super(CategoricalIndex, self).__init__(values, **kwargs)
-        assert self._values.null_count == 0
```

Comment:
It seems that dask assumes it can initialize a `CategoricalIndex` with null values